test/mpi/test_allgather.cc

```diff
@@ -19,21 +19,30 @@ TestAllgather::TestAllgather(size_t _msgsize, ucc_test_mpi_inplace_t _inplace,
     MPI_Comm_rank(team.comm, &rank);
     MPI_Comm_size(team.comm, &size);
 
+    args.coll_type         = UCC_COLL_TYPE_ALLGATHER;
+
+    if (skip(test_max_size &&
+              (test_max_size < (_msgsize*size)),
+        TEST_SKIP_MEM_LIMIT, team.comm)) {
+        return;
+    }
+
     UCC_CHECK(ucc_mc_alloc(&rbuf, _msgsize*size, _mt));
-    UCC_CHECK(ucc_mc_alloc(&check_buf, _msgsize*size, _mt));
+    UCC_CHECK(ucc_mc_alloc(&check_rbuf, _msgsize*size, UCC_MEMORY_TYPE_HOST));
     if (TEST_NO_INPLACE == inplace) {
         UCC_CHECK(ucc_mc_alloc(&sbuf, _msgsize, _mt));
         init_buffer(sbuf, count, TEST_DT, _mt, rank);
+        UCC_CHECK(ucc_mc_alloc(&check_sbuf, _msgsize, UCC_MEMORY_TYPE_HOST));
+        init_buffer(check_sbuf, count, TEST_DT, UCC_MEMORY_TYPE_HOST, rank);
```

Comment:
why not to use UCC_ALLOC_COPY_BUF ?  we just need to copy from check_sbuf to sbuf
same question for other tests

Reply:
fixed